mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -172,16 +174,25 @@ public List<MessageResult> getMessages(Collection<MessageId> messageIds, FetchGr
 
     @Override
     public Flux<MessageResult> getMessagesReactive(Collection<MessageId> messageIds, FetchGroup fetchGroup, MailboxSession mailboxSession) {
+        // Here
         MessageIdMapper messageIdMapper = mailboxSessionMapperFactory.getMessageIdMapper(mailboxSession);
 
         MessageMapper.FetchType fetchType = FetchGroupConverter.getFetchType(fetchGroup);
         return messageIdMapper.findReactive(messageIds, fetchType)
             .groupBy(MailboxMessage::getMailboxId)
             .filterWhen(groupedFlux -> hasRightsOnMailboxReactive(mailboxSession, Right.Read).apply(groupedFlux.key()), DEFAULT_CONCURRENCY)
             .flatMap(Function.identity(), DEFAULT_CONCURRENCY)
+            .publishOn(forFetchType(fetchType))
             .map(Throwing.function(messageResultConverter(fetchGroup)).sneakyThrow());
     }
 
+    private Scheduler forFetchType(MessageMapper.FetchType fetchType) {
```

Comment:
Duplicate. Do you think this could have its place in `MailboxReactorUtils` for example?

Not a big deal though